relay-general/src/store/transactions/rules.rs

```diff
@@ -111,7 +111,7 @@ pub struct TransactionNameRule {
 
 impl TransactionNameRule {
     /// Checks is the current rule matches and tries to apply it.
-    pub fn match_and_apply(
+    pub fn transaction_match_and_apply(
```

Comment:
I don't really like this (especially the code duplication), but it's the fastest way I could think of to make the `TransactionNameRule` generic enough for this PR without spending time in a refactor to make a clean interface.

Reply:
You could get rid of code duplication by making `transaction_info` an optional argument to `match_and_apply` (and only checking the transaction source if it's `Some`).

Reply:
I'll refactor this at some point to extract common functionality into a trait, and both the API and logic should get much easier to read and deal with.